### PR TITLE
update revoked_token id field type

### LINKED ISSUE #662: Type error. storing/retrieving string as an id
**Describe the bug**
We have been using sqllite which doesn't enforce types. So this bug goes unnoticed.

Using postgres as the database the following error occurs
```python
self = <sqlalchemy.dialects.postgresql.psycopg2.PGDialect_psycopg2 object at 0x113ef6400>, cursor = <cursor object at 0x113b11220; closed: -1>
statement = 'SELECT revoked_tokens.created_at AS revoked_tokens_created_at, revoked_tokens.updated_at AS revoked_tokens_updated_at...revoked_tokens_id, revoked_tokens.jti AS revoked_tokens_jti \nFROM revoked_tokens \nWHERE revoked_tokens.id = %(pk_1)s'
parameters = {'pk_1': '7a2ab50d-8d3e-4278-9a18-ec65c1cfe879'}, context = <sqlalchemy.dialects.postgresql.psycopg2.PGExecutionContext_psycopg2 object at 0x113f47ac0>

    def do_execute(self, cursor, statement, parameters, context=None):
>       cursor.execute(statement, parameters)
E       sqlalchemy.exc.DataError: (psycopg2.errors.InvalidTextRepresentation) invalid input syntax for type integer: "7a2ab50d-8d3e-4278-9a18-ec65c1cfe879"
E       LINE 3: WHERE revoked_tokens.id = '7a2ab50d-8d3e-4278-9a18-ec65c1cfe...
E                                         ^
E
E       [SQL: SELECT revoked_tokens.created_at AS revoked_tokens_created_at, revoked_tokens.updated_at AS revoked_tokens_updated_at, revoked_tokens.id AS revoked_tokens_id, revoked_tokens.jti AS revoked_tokens_jti
E       FROM revoked_tokens
E       WHERE revoked_tokens.id = %(pk_1)s]
E       [parameters: {'pk_1': '7a2ab50d-8d3e-4278-9a18-ec65c1cfe879'}]
E       (Background on this error at: https://sqlalche.me/e/14/9h9h)

../../../.local/share/virtualenvs/dwellinglybackend-S6sCqKnp/lib/python3.8/site-packages/sqlalchemy/engine/default.py:717: DataError
```

**To Reproduce**
Run this test with a postgresql database:
- `tests/integration/test_users.py::TestUserLogin::test_refresh_user`

**Expected behavior**
Test should pass.

### PR DESCRIPTION
**Pull request by qq1u**
### What issue is this solving?
fix 662

Closes codeforpdx/dwellingly-app/issues/662

### CODE REVIEW

**Comment by NickSchimek** [comment]
Yeah, This change will work. I don't think we need the jti column anymore if we're going to store the jti as the id of the table.

I'm a bit hesitant to approve this without checking how this model is used in the application. It seems the original implementation meant for the jti column to be used, but obviously that is not what's being done. Perhaps this change is Okay though. Did you find anything to suggest otherwise?

**Comment by qq1u** [comment]
> Yeah, This change will work. I don't think we need the jti column anymore if we're going to store the jti as the id of the table.
> 
> I'm a bit hesitant to approve this without checking how this model is used in the application. It seems the original implementation meant for the jti column to be used, but obviously that is not what's being done. Perhaps this change is Okay though. Did you find anything to suggest otherwise?



I think the original meaning of the program is to blocktlist n the future
However, there is currently no move to blacklist the token in the program.

File: models/revoked_tokens.py [hunk 1]
```diff
@@ -5,7 +5,7 @@
 class RevokedTokensModel(BaseModel):
     __tablename__ = "revoked_tokens"
 
-    id = db.Column(db.Integer, primary_key=True)
+    id = db.Column(db.String(36), primary_key=True)
```
Comment (NickSchimek) [inline models/revoked_tokens.py]: I don't think the solution is to change the id field to a String. That does make the test pass, but will cause other issues that we are apparently missing tests for.

**Review by NickSchimek — CHANGES_REQUESTED** [review]
When a user logs out we should store their jwt in this table, unfortunately there is no logic that I can see that handles logging out a user. I'm not sure how all this was initially setup with the frontend, but that does seem incorrect.

I don't think changing the column types to make the test pass is the correct solution.

If you look at app.py you will find the following code which calls `is_jwt_blacklisted`

https://github.com/codeforpdx/dwellinglybackend/blob/6307272acdf11ef542f0ddd9b96fc363855ac45c/app.py#L48-L51

And if you look at that method you might be able to see why there is an error:

https://github.com/codeforpdx/dwellinglybackend/blob/6307272acdf11ef542f0ddd9b96fc363855ac45c/models/revoked_tokens.py#L11-L13

It is callling the method `get` with the jwt string. get is not what we want to use here as it looks up by the id column. But we want to find a row in the table by the jwt column.

**Pull request merged**